### PR TITLE
feat(tooltip): add rh-tooltip

### PR DESCRIPTION
**Pull request by brianferry**
## What I did

Added `rh-tooltip` to RHDS based off of the `BaseTooltip` class from patternfly elements https://github.com/patternfly/patternfly-elements/tree/main/elements/pfe-tooltip. 

1. Updated the styles to match [rh-tooltip](https://ux.redhat.com/components/tooltip/)
2. Updated documentation and tests to match rh-tooltip expected values (top, right, bottom, left positions) 

## Testing Instructions

1. git checkout origin/feat/rh-global-footer
2. npm ci
3. npm run dev
4. Visit http://localhost:8000/components/tooltip/demo/

## Notes to Reviewers

1. The main updates present in this file are related to the styles in the tooltip as all of the functionality is located in the `BaseTooltip` from Patternfly Elements.  The only exception to this is the dark theme colors present for rh-tooltip and not pfe-tooltip.

### CODE REVIEW

**Comment by brianferry** [comment]
@marionnegp - Would you be available to take a look at the demo at https://deploy-preview-465--red-hat-design-system.netlify.app/components/tooltip/demo/ and confirm that it matches https://ux.redhat.com/components/tooltip/ ?

**Comment by marionnegp** [comment]
- [ ] Padding is showing up as 14px, but I see that you have it set to 1em. Should be 16px.
- [ ] Arrow size should be 16px wide x 9px tall.
- [ ] Space between the top and bottom tooltips and their buttons is 10px. Should be 12px. (Left and right tooltips look right!)

I don't think we specified a max-width in our docs, but what you have looks good to me.

File: elements/rh-tooltip/rh-tooltip.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import { pfelement, colorContextConsumer } from '@patternfly/pfe-core/decorators.js';
+import { BaseTooltip } from '@patternfly/pfe-tooltip/BaseTooltip.js';
+
+import styles from './rh-tooltip.css';
+
+/**
+ * Tooltip
+ * @slot - Place element content here
+ */
+@customElement('rh-tooltip') @pfelement()
```
Comment (bennypowers) [inline elements/rh-tooltip/rh-tooltip.ts:13]:
```suggestion
@customElement('rh-tooltip')
```

File: elements/rh-tooltip/rh-tooltip.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import type { ColorTheme } from '@patternfly/pfe-core';
+import { customElement, property } from 'lit/decorators.js';
+
+import { pfelement, colorContextConsumer } from '@patternfly/pfe-core/decorators.js';
```
Comment (bennypowers) [inline elements/rh-tooltip/rh-tooltip.ts:4]:
```suggestion
import { colorContextConsumer } from '@patternfly/pfe-core/decorators.js';
```

**Review by bennypowers — CHANGES_REQUESTED** [review]

File: elements/rh-tooltip/rh-tooltip.css [hunk 3]
```diff
@@ -0,0 +1,14 @@
+:host([on=dark]) {
+  --pf-c-tooltip__content--BackgroundColor: #ffffff;
+  --pf-c-tooltip__content--Color: #151515;
+}
+
+:host {
+  --pf-c-tooltip--BoxShadow: 0 0.125rem 0.25rem 0 rgba(21, 21, 21, 0.2);
```
Comment (bennypowers) [inline elements/rh-tooltip/rh-tooltip.css:7]: reserving this comment pending @marionnegp's design review.

Resolve this (and any subsequent) comment threads to auto-merge

**Review by bennypowers — APPROVED** [review]

**Comment by marionnegp** [comment]
I double checked the spacing between the tooltip and the invoker, and it's actually supposed to be 8px. Sorry for saying it was 10px! It looks like the spacing is only 1px off for the tooltips, but I couldn't tell without taking screenshots and zooming in. 

The tooltip looks good to me.

**Comment by brianferry** [comment]
@marionnegp - No worries at all, should be fixed now.

**Review by marionnegp — APPROVED** [review]

**Comment by bennypowers** [comment]
Congrats on the merge @brianferry

**Pull request merged**